commons-core/src/main/scala/com/avsystem/commons/serialization/json/JsonStringOutput.scala

```diff
@@ -44,17 +44,17 @@ final class JsonStringOutput(builder: JStringBuilder) extends BaseJsonOutput wit
   def writeString(str: String): Unit = writeJsonString(builder, str)
   def writeBoolean(boolean: Boolean): Unit = builder.append(boolean.toString)
   def writeInt(int: Int): Unit = builder.append(int.toString)
-  def writeLong(long: Long): Unit = {
-    val asInt = long.toInt
-    if (asInt == long) writeInt(asInt)
-    else writeString(long.toString)
+  def writeLong(long: Long): Unit = builder.append(long.toString)
+  def writeDouble(double: Double): Unit = {
+    if (java.lang.Double.isNaN(double) || double == Double.NegativeInfinity || double == Double.PositiveInfinity)
+      writeString(double.toString)
+    else builder.append(double.toString)
   }
-  def writeDouble(double: Double): Unit = builder.append(double.toString)
   def writeBinary(binary: Array[Byte]): Unit = {
     builder.append('"')
     var i = 0
     while (i < binary.length) {
-      builder.append(f"${binary(i)}%02x")
+      builder.append(f"${binary(i) & 0xff}%02x") //JS has signed chars
```

Comment:
What was wrong with the previous version?
I don't care about chars in JS. It's JSON, not JS and I'm just representing Scala bytes in hex. Hex is "unsigned" but I don't care about it as long as it deserializes properly.

Reply:
It produces a different string representation which won't even deserialize with current, logical implementation of an Input. 
```scala
val binary: Byte = -2
val string = f"${binary}%02x" //"fffffffe" in SJS and "fe" on JVM
```
I do care about JS compat and you also should if you want this to be used for e.g. Udash RPCs.

Reply:
That is a ScalaJS bug for me. The fact that it works like this in JS is not an excuse.
But I'm fine with the workaround. I checked that only on JVM.

Reply:
It is a bug in the sense that it works differently than on the JVM, but it's not a bug if you just read the docs. %02x is "at least two hexes", so with signed chars this makes sense. It would work exactly the same on a machine with signed char in plain old C.